File: Activity/app/src/main/java/ru/activity/activity/MainActivity.java
package ru.activity.activity;

import android.nfc.Tag;
import android.util.Log;
import android.os.Bundle;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private final static String TAG = "Активити";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        String instanceState;
        if (savedInstanceState == null)
        {
            instanceState = "Первый запуск";
        }
        else {
            instanceState = "Второй запуск";
        }
        Toast.makeText(getApplicationContext(), " - onCreate", Toast.LENGTH_SHORT).show();

       Log.i(TAG,"onCreate");

    }



    @Override
    protected void onStart()
    {
        super.onStart();
        Toast.makeText(getApplicationContext(), " - onStart", Toast.LENGTH_SHORT).show();
        Log.i(TAG,"onStart");
    }

  @Override
  protected void onRestoreInstanceState(Bundle savedInstanceState )
  {
      super.onRestoreInstanceState(savedInstanceState);
      Toast.makeText(getApplicationContext(), " - Поворный запуск - onRestoreInstanceState ", Toast.LENGTH_SHORT).show();
      Log.i(TAG,"onRestoreInstanceState");
  }
    @Override
    protected void onResume()
    {
        super.onResume();
        Toast.makeText(getApplicationContext(), " - onResume", Toast.LENGTH_SHORT).show();
        Log.i(TAG,"onResume");
    }
    @Override
    protected void onPause(){
        super.onPause();
        Toast.makeText(getApplicationContext(), " - onPause", Toast.LENGTH_SHORT).show();
        Log.i(TAG,"onPause");
    }
    @Override
    protected void onSaveInstanceState(Bundle savedInstanceState){
        super.onSaveInstanceState(savedInstanceState);
        Toast.makeText(getApplicationContext(), " - onSaveInstanceState", Toast.LENGTH_SHORT).show();
        Log.i(TAG,"onSaveInstanceState");
    }
    @Override
    protected void onStop(){
        super.onStop();
        Toast.makeText(getApplicationContext(), " - onStop", Toast.LENGTH_SHORT).show();
        Log.i(TAG,"onStop");
    }
    @Override
    protected void onRestart(){
        super.onRestart();
        Toast.makeText(getApplicationContext(), " - onRestart", Toast.LENGTH_SHORT).show();
        Log.i(TAG,"onRestart");
    }
    @Override
    protected void onDestroy(){
        super.onDestroy();
        Toast.makeText(getApplicationContext(), " - onDestroy", Toast.LENGTH_SHORT).show();
        Log.i(TAG,"onDestroy");
    }
}

File: app/src/main/java/ru/home/k_lesson_1/MainActivity.kt
package ru.home.k_lesson_1

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View

class MainActivity : AppCompatActivity() {
    //свойства(поля)
    companion object Res{
        @JvmStatic
        val field1 = "STATIC_STRING"
    }

    enum class WeatherType {
        SUNNY,
        RAINY,
        CLOUDY,
        MISTY,
        SNOWY,
        HAILY
    }


    val  listener = object : View.OnClickListener{
        override fun onClick(v: View?) {
            TODO("Not yet implemented")
        }

    }


    var i:Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val objTest = Test("name")
        Log.d("mylog", "${objTest.name}")

        //анонимний класс - класс которого не существует, нет имени. Экземпляр анонимного класса. Можем обратиться его экземпляру, к самому анниму
        //обратиться нельзя
        val test3 = object {
            var i:Int=0
            val field1 = "STATIC_STRING"
        }
        Log.d("mylog", "${test3.i}")
        Log.d("mylog", "${test3.field1}")

        val weatherType:WeatherType = WeatherType.CLOUDY
        this.i=when(weatherType){
            WeatherType.CLOUDY-> 1
            WeatherType.HAILY-> 2
            WeatherType.MISTY-> 3
            WeatherType.RAINY-> 4
            WeatherType.SNOWY-> 5
            WeatherType.SUNNY-> 6
        }

        this.i=1
        val str= when(this.i){
           1->"1"
            2->{val int1:Int=12
            int1}
            else->1
        }

        Log.d("mylog", "$str")


//        this.i=if(objTest.i==3) 2 else 3





        val getI=objTest.i
        objTest.i =2
        this.i=2
        this.i=try {
            throw IllegalStateException()
        }catch (e:IllegalStateException)
        {3}
            Log.d("mylog", "$i")

    }
}
class Test(var name:String=" ")
{
        var i:Int=0
}
fun test():Int{
    var i:Int =0
    return i
}